checker/tests/index/BinarySearchTest.java

```diff
@@ -0,0 +1,43 @@
+import java.util.Arrays;
+import org.checkerframework.checker.index.qual.*;
+
+//@skip-test because of a samelen issue
```

Comment:
This was the viewpoint adaptation issue where I wasn't creating a `DependentTypesTreeAnnotator`. I forgot that it was impacting this PR too. I've reenabled the test and checked that it passes now (at least locally).